ores/scoring_systems/celery_queue.py

```diff
@@ -0,0 +1,287 @@
+                           revscoring.errors.UserNotFound,
+                           revscoring.errors.DependencyError,
+                           mwapi.errors.RequestError,
+                           mwapi.errors.TimeoutError,
+                           TimeoutError)
+
+        @self.application.task(throws=expected_errors,
+                               queue=DEFAULT_CELERY_QUEUE,
+                               priority=0)
+        def _lookup_score_in_map(result_id, model_name):
+            logger.info("Looking up {0} in {1}".format(model_name, result_id))
+            score_map_result = self._process_score_map.AsyncResult(result_id)
+            try:
+                score_map = score_map_result.get(timeout=self.timeout)
+            except Exception as e:
+                logger.error(e)
+                raise
+            logger.info("Found up {0} in {1}!".format(model_name, result_id))
+            return score_map[model_name]
+
+        @self.application.task(throws=expected_errors,
+                               queue=DEFAULT_CELERY_QUEUE,
+                               priority=0)
+        def _process_score_map(
+             context_name, model_names, rev_id, root_cache=None,
+             injection_cache=None, include_features=False):
+            logger.info("Generating a score map for" +
+                        "{0}:{1}:{2}"
+                        .format(context_name, set(model_names), rev_id))
+            try:
+                score_map = ScoringSystem._process_score_map(
+                    self, context_name, model_names, rev_id,
+                    root_cache=root_cache,
+                    injection_cache=injection_cache,
+                    include_features=include_features)
+            except Exception as e:
+                logger.error(e)
+                raise
+            logger.info("Completed generating score map for" +
+                        "{0}:{1}:{2}"
+                        .format(context_name, set(model_names), rev_id))
+            return score_map
+
+        APPLICATIONS.append(application)
+
+        self._process_score_map = _process_score_map
+        self._lookup_score_in_map = _lookup_score_in_map
+
+    def _process_missing_scores(self, context_name, missing_model_set_revs,
+                                root_caches, injection_caches,
+                                include_features, inprogress_results=None):
+        logger.debug("Processing missing scores {0}:{1}."
+                     .format(context_name, missing_model_set_revs))
+        context = self[context_name]
+
+        inprogress_results = inprogress_results or {}
+
+        # Generate score results
+        results = {}
+        for missing_models, rev_ids in missing_model_set_revs.items():
+            for rev_id in rev_ids:
+                injection_cache = injection_caches.get(rev_id) \
+                                  if injection_caches is not None else None
```

Comment:
This clause is repeated many times. What is the reason for this complexity / flexibility?